packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -515,53 +516,46 @@ function toCppNamespace(namespace: string) {
   return namespace.replace(/\./g, '::');
 }
 
-export function installDependencies(options: {verbose: boolean}) {
-  const cwd = process.cwd();
-
-  // Extract react-native peer dependency version
-  const rnwPackageJsonPath = require.resolve(
-    'react-native-windows/package.json',
-    {paths: [process.cwd()]},
-  );
-  const rnwPackageJson = JSON.parse(
-    fs.readFileSync(rnwPackageJsonPath, {encoding: 'UTF8'}),
-  );
-  let rnPeerDependency = rnwPackageJson.peerDependencies['react-native'];
-  const depDelim = ' || ';
-  const delimIndex = rnPeerDependency.indexOf(depDelim);
-  if (delimIndex !== -1) {
-    rnPeerDependency = rnPeerDependency.slice(0, delimIndex);
+export async function installScriptsAndDependencies(options: {
+  verbose: boolean;
+}) {
+  const projectPackage = await WritableNpmPackage.fromPath(process.cwd());
+  if (!projectPackage) {
```

Comment:
below the code usses null propagation operator on the json field, should we check that here as well to fail early?

Reply:
Oh whoops, was just meaning to do that for the dependencies prop instead of all the JSON. Will fix that.